lib/state/index.ts

```diff
@@ -50,6 +50,20 @@ export type State = {
 
 const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
 
+const actionGeneratorMiddleware: Middleware = (store) => (
+  next: (action: A.ActionType) => A.ActionType
+) => (action: A.ActionType | ReturnType<ActionGenerator>) => {
+  const state = store.getState();
+  if (typeof action === 'function') {
+    const result = action(state);
+    if (!result || !result.type) {
+      return;
+    }
```

Comment:
This gives control to the action generator to prevent the action dispatch, it could be useful if for example we determine that the Redux action shouldn't be dispatched.